Plugin/FakeIpPlugin.php

```diff
@@ -48,7 +55,14 @@ public function handleQuery(Query $query, callable $next, callable $first)
             return $next($query);
         }
 
-        $text = str_replace($this->needle, $this->replacement, $query->getText(), $count);
+        $replacement = $this->replacement;
+
+        if ($this->useFaker) {
+            $faker = Factory::create();
```

Comment:
I have only concern that for each query the factory will create a new generator.
Also if we are using only the `internet` data provider, we could instantiate a generator only with this provider, wdyt? :)

Reply:
good point! I will work on it! :)

Reply:
What do yo thin about this and the best way to check if plugin needs to use faker?

```
    /**
     * @var Generator
     */
    private $faker;

    public function __construct(string $needle, string $replacement = null, bool $useFaker = false)
    {
        $this->needle = $needle;
        $this->replacement = $replacement;
        $this->useFaker = $useFaker;

        if ($useFaker) {
            $this->faker = new Generator();
            $this->faker->addProvider(new Internet($this->faker));
        }
    }

    /**
     * {@inheritdoc}
     */
    public function handleQuery(Query $query, callable $next, callable $first)
    {
        if (!$query instanceof GeocodeQuery) {
            return $next($query);
        }

        $replacement = $this->replacement;

        if ($this->useFaker) { // OR null !== $this->faker ?
            $replacement = $this->faker->ipv4;
        }
```

Reply:
Yes, this looks nice, just add a `Generator|null` typehint on `$faker` so sca tools won't complain and use the `null !== $this->faker` check :)